### PR TITLE
Added soft descriptor to merchant

### PR DESCRIPTION
**Pull request by nilssonemma**

### CODE REVIEW

File: Merchant/index.ts [hunk 1]
```diff
@@ -34,7 +34,11 @@ export namespace Merchant {
 			(value.type == "test" || value.type == "live") &&
 			typeof value.agent == "string" &&
 			typeof value.reference == "string" &&
-			(value.descriptor == undefined || typeof value.descriptor == "string") &&
+			(value.descriptor == undefined ||
+				typeof value.descriptor == "string" ||
+				(typeof value.descriptor == "object" &&
```
Comment (robinnedergard) [inline Merchant/index.ts:39]:
```suggestion
				(typeof value.descriptor == "object" &&
				value.descriptor &&
```

Comment (robinnedergard) [reply Merchant/index.ts:39]: avoid null

File: Merchant/index.ts [hunk 2]
```diff
@@ -58,7 +62,13 @@ export namespace Merchant {
 							typeof value.agent == "string" || { property: "agent", type: "string" },
 							typeof value.reference == "string" || { property: "reference", type: "string" },
 							value.descriptor == undefined ||
-								typeof value.descriptor == "string" || { property: "descriptor", type: "string | undefined" },
+								typeof value.descriptor == "string" ||
+								(typeof value.descriptor == "object" &&
```
Comment (robinnedergard) [inline Merchant/index.ts:66]:
```suggestion
								(typeof value.descriptor == "object" &&
								value.descriptor &&
```

File: Merchant/index.ts [hunk 3]
```diff
@@ -71,6 +81,9 @@ export namespace Merchant {
 					  ].filter(gracely.Flaw.is) as gracely.Flaw[]),
 		}
 	}
+	export function getDecriptor(merchant: Merchant): string | undefined {
```
Comment (robinnedergard) [inline Merchant/index.ts:84]:
```suggestion
	export function getDescriptor(merchant: Merchant): string | undefined {
```

**Review by robinnedergard — CHANGES_REQUESTED** [review]
Also: State/Merchant.ts#L47 github actions report a type error

**Review by fredrikbryntesson — APPROVED** [review]

**Pull request merged**